src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -1665,6 +1665,15 @@ private void HandleDirective(DirectiveDescriptor descriptor)
                     {
                         Span.ChunkGenerator = SpanChunkGenerator.Null;
                         Output(SpanKindInternal.Code, AcceptedCharactersInternal.WhiteSpace);
+
+                        if (EndOfFile || At(CSharpSymbolType.NewLine))
+                        {
+                            // Add a marker symbol to provide CSharp intellisense.
```

Comment:
More specific in the comment so when we come back and revisit the hack we'll know 😄